app/src/main/java/io/github/sumukhshiv/mdbsocials/NewSocialActivity.java

```diff
@@ -0,0 +1,210 @@
+        firebaseUser = firebaseAuth.getCurrentUser();
+
+
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+
+        //Detects request codes
+        //Retrieves image onActivityResult and sets it to the imageView
+        if((requestCode==GET_FROM_GALLERY || requestCode == REQUEST_CAMERA) && resultCode == Activity.RESULT_OK) {
+            Uri selectedImage = data.getData();
+            Bitmap bitmap = null;
+            try {
+                bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), selectedImage);
+                //BitmapDrawable bitmapDrawable = new BitmapDrawable(getResources(), bitmap);
+                ((ImageView) findViewById(R.id.imageViewUploadPhoto)).setImageBitmap(bitmap);
+            } catch (FileNotFoundException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            } catch (IOException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+        }
+
+        this.data = data;
+    }
+
+    @Override
+    public void onClick(View view) {
+        switch (view.getId()) {
+            //Adding a photo case
+            case (R.id.buttonUploadPicture):
+                AlertDialog alertDialog = new AlertDialog.Builder(NewSocialActivity.this).create();
+                alertDialog.setTitle("Set a Photo");
+                alertDialog.setButton(AlertDialog.BUTTON_POSITIVE, "Take a Photo",
+                        new DialogInterface.OnClickListener() {
+                            public void onClick(DialogInterface dialog, int which) {
+                                //Open Camera
+                                Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+                                if (takePictureIntent.resolveActivity(getPackageManager()) != null) {
+                                    startActivityForResult(takePictureIntent, REQUEST_CAMERA);
+                                }
+                                dialog.dismiss();
+                            }
+                        });
+                alertDialog.setButton(AlertDialog.BUTTON_NEGATIVE, "Upload from Gallery",
+                        new DialogInterface.OnClickListener() {
+                            public void onClick(DialogInterface dialog, int which) {
+                                //launch gallery
+                                startActivityForResult(new Intent(Intent.ACTION_PICK, android.provider.MediaStore.Images.Media.INTERNAL_CONTENT_URI), GET_FROM_GALLERY);
+                                dialog.dismiss();
+                            }
+                        });
+                alertDialog.show();
+                break;
+
+            //Posting the Social Firebase and Newsfeed
+            case (R.id.buttonPost):
```

Comment:
Having so much code below this case makes it hard to read. I recommend modularizing it into a function